cluster/cluster.go

```diff
@@ -114,32 +114,27 @@ func PeersForQuery() []Node {
 	answer := make([]Node, 0)
 	// we want to get the minimum number of nodes
 	// needed to cover all partitions
+
+LOOP:
 	for _, nodes := range peersMap {
-		selected := nodes[0]
 		// always prefer the local node which will be nodes[0]
 		// if it has this partition
-		if selected.Name != thisNode.Name {
-			// check if we are already going to use one of the
-			// available nodes and re-use it
-			reusePeer := false
-			for _, n := range nodes {
-				if _, ok := selectedPeers[n.Name]; ok {
-					selected = n
-					reusePeer = true
-					break
-				}
-			}
-			// if no nodes have been selected yet then grab a
-			// random node from the set of available nodes
-			if !reusePeer {
-				selected = nodes[rand.Intn(len(nodes))]
-			}
+		if nodes[0].Name == thisNode.Name {
+			answer = append(answer, selected)
+			continue LOOP
```

Comment:
nice! now that you found a way to move this out of the inner loop, the label for outer loop actually wouldn't be necessary anymore